wasmtime-jni/src/wtrap.rs

```diff
@@ -21,3 +21,26 @@ pub fn from_java(env: &JNIEnv, obj: JObject) -> Result<Trap> {
         _ => return Err(Error::UnknownEnum(name)),
     })
 }
+
+pub fn into_java<'a>(env: &JNIEnv<'a>, trap: &Trap) -> jni::errors::Result<JObject<'a>> {
+    if let Some(status) = trap.i32_exit_status() {
+        Ok(env
+            .call_static_method(
+                "io/github/kawamuray/wasmtime/Trap",
+                "fromExitCode",
+                "(I)Lio/github/kawamuray/wasmtime/Trap;",
+                &[status.into()],
+            )?
+            .l()?)
+    } else {
+        let jmsg = env.new_string(trap.to_string())?;
```

Comment:
Yeah, I feel the same thing. 😓  Unfortunately, [`TrapReason`](https://github.com/bytecodealliance/wasmtime/blob/63d5b919306ebecc00cd39090910d89c02dcda9b/crates/wasmtime/src/trap.rs#L17) is an internal type, so there is no way to distinguish `Message` and `Error` cases.
But `to_string()` method uses `fmt::Display` implementation of the type and [the impl doesn't write backtrace](https://github.com/bytecodealliance/wasmtime/blob/63d5b919306ebecc00cd39090910d89c02dcda9b/crates/wasmtime/src/trap.rs#L28-L36), so it would not be so surprising

Reply:
>  TrapReason is an internal type,

ah right, sorry I missed that.

Yeah apparently calling `to_string()` is the only possible way to get data that contains `message` given for the trap... nevermind then.